osu.Framework/Graphics/Sprites/SpriteText.cs

```
@@ -155,59 +156,73 @@ public float TextSize
         }
 
         [BackgroundDependencyLoader]
-        private void load()
+        private void load(ILocalisationEngine localisation)
         {
+            localisationEngine = localisation;
+            updateLocalisedBindable();
+
             spaceWidth = CreateCharacterDrawable('.')?.DrawWidth * 2 ?? default_text_size;
 
             validateLayout();
         }
 
-        private Bindable<string> current;
+        private ILocalisationEngine localisationEngine;
+
+        [NotNull]
+        private string displayText = string.Empty;
+
+        private LocalisableString localisableText = new LocalisableString(string.Empty, LocalisationType.None);
 
         /// <summary>
-        /// Implements the <see cref="IHasCurrentValue{T}"/> interface.
+        /// Get or set a <see cref="LocalisableString"/> providing all necessary information to apply localisation to a text sequence.
+        /// <para>Changing the bindable values is allowed, although you should create a new <see cref="LocalisableString"/> when changing two or more values.</para>
         /// </summary>
-        public Bindable<string> Current
+        [NotNull]
+        public LocalisableString LocalisableText
         {
-            get => current;
+            get => localisableText;
             set
             {
```

Should throw an exception if `null` is provided.

---

Also could do with a `[NotNull]` attribute